BikeData/src/main/scala/by/itechart/internship/Main.scala

```diff
@@ -1,20 +1,27 @@
 package by.itechart.internship
 
-import scala.util.{Failure, Success, Try}
+import by.itechart.internship.config.{FlywayConfig, LightBendConfig}
+import by.itechart.internship.dao.DatabaseReader
+import by.itechart.internship.logic._
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+import scala.concurrent.{Await, Future}
 
 object Main extends App {
   val configValues = LightBendConfig.configSetter
-  Try(FileReader.tableReader(configValues)) match {
-    case Success(data) => {
-      val dataTableOfTrips = data.getLines()
-        .map(_.replaceAll("\"", "").split(configValues.delimiterOfFile))
-        .toList
-        .drop(configValues.nameColumnsIndex)
+  val flywayConfig = FlywayConfig.initDatabaseStructure(configValues)
 
-      GeneralStats.logicController(configValues, dataTableOfTrips)
-      UsageStats.logicController(configValues, dataTableOfTrips)
-      BikeStats.logicController(configValues, dataTableOfTrips)
+  val result = FileReader.tableReader(configValues).map(_ => {
+    val dataTrips = DatabaseReader.writer().map{vector =>
+      val listOfRecords = List(
+        GeneralStats.logicController(configValues, vector),
+        UsageStats.logicController(configValues, vector),
+        BikeStats.logicController(configValues, vector))
+      FileWriter.tableWriter(configValues, listOfRecords)
     }
-    case Failure(e) => println(s"An error has occured, cause: $e")
-  }
+    Await.result(dataTrips, 720.seconds)
```

Comment:
You don't need this await
I guess your problem is that without await your type becomes Future[Future[Result]] and main app exits before inner Future[Result] completes
Instead of await, just use FileReader.tableReader(configValues).flatMap instead of FileReader.tableReader(configValues).map

FlatMap will flatten result, so it will be not Future[Future[A]], but just Future[A], which makes sense